src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockDepths/boss_magmus.cpp

```diff
@@ -110,7 +117,61 @@ class boss_magmus : public CreatureScript
         }
 };
 
+class npc_ironhand_guardian : public CreatureScript
+{
+    public:
+
+        npc_ironhand_guardian() : CreatureScript("npc_ironhand_guardian") { }
+
+        struct npc_ironhand_guardianAI : public ScriptedAI
+        {
+
+            npc_ironhand_guardianAI(Creature* creature) : ScriptedAI(creature)
+            {
+                _instance = me->GetInstanceScript();
+                _lastState = 0;
+            }
+
+            void Reset() override
+            {
+                _events.Reset();
```

Comment:
dont need reset _lastState here?

Reply:
@martintopholm

Reply:
@martintopholm should Reset() set _lastState back to 0 ?

Reply:
No it shouldn't reset _lastState. Once they're activated, they keep casting flames until the next instance reset. Though it needs a better name and type (bool) to reflect its behavior.